Modules/Devices/src/Devices.Application/Clients/Commands/ChangeClientSecret/Handler.cs

```diff
@@ -16,10 +16,9 @@ public async Task<ChangeClientSecretResponse> Handle(ChangeClientSecretCommand r
         var client = await _oAuthClientsRepository.Find(request.ClientId, cancellationToken);
 
         var clientSecret = string.IsNullOrEmpty(request.NewSecret) ? PasswordGenerator.Generate(30) : request.NewSecret;
-        client.ClientSecret = clientSecret;
 
-        await _oAuthClientsRepository.Update(client, cancellationToken);
+        await _oAuthClientsRepository.ChangeClientSecret(client, clientSecret, cancellationToken);
```

Comment:
I'd suggest that we change the `ChangeClientSecret` method to accept clientId and the new secret. Then we don't need to read the client only to pass it back to the repository again. This seems kind of unnecessary.

Reply:
We need the client for the ChangeClientSecretResponse. Should the handler only return the ClientId and ClientSecret instead of all the OAuthClient fields?

Reply:
Hmm. Okay, ignore my comment. :)